vehicle/mkv/software/air_control/tests/test_air.py

```diff
@@ -0,0 +1,413 @@
+MC_TAU = 0.2
+
+
+def reset(iocontroller, pins):
+    iocontroller.write_pin(pins["RESET"][0], PinValue.LOW)
+    time.sleep(0.5)
+    iocontroller.write_pin(pins["RESET"][0], PinValue.HIGH)
+
+
+def initialize_to_idle(canbus, iocontroller, pins):
+    canbus.set_state("pack_voltage", 350)
+    canbus.set_periodic("bms_core", 0.1)
+    canbus.set_state("D1_DC_Bus_Voltage", 0)
+    canbus.set_periodic("M167_Voltage_Info", 0.1)
+
+    iocontroller.write_pin(pins["AIR_P_WELD_DETECT"][0], PinValue.LOW)
+    iocontroller.write_pin(pins["AIR_N_WELD_DETECT"][0], PinValue.LOW)
+    iocontroller.write_pin(pins["SS_TSMS"][0], PinValue.HIGH)
+    iocontroller.write_pin(pins["IMD_SENSE"][0], PinValue.HIGH)
+    iocontroller.write_pin(pins["BMS_SENSE"][0], PinValue.HIGH)
+
+
+"""
+Tests
+"""
+
+
+def test_idle_bms_fault(canbus, iocontroller, pins):
+    initialize_to_idle(canbus, iocontroller, pins)
+    reset(iocontroller, pins)
+    time.sleep(0.5)
+    assert canbus.get_state("air_state") == "IDLE"
+
+    # IMD FAULT
+    iocontroller.write_pin(pins["IMD_SENSE"][0], PinValue.LOW)
+    time.sleep(0.1)
+    assert canbus.get_state("air_state") == "FAULT"
+    assert canbus.get_state("air_fault") == "IMD_STATUS"
+
+
+def test_idle_imd_fault(canbus, iocontroller, pins):
+    initialize_to_idle(canbus, iocontroller, pins)
+    reset(iocontroller, pins)
+    time.sleep(0.5)
+    assert canbus.get_state("air_state") == "IDLE"
+
+    # IMD FAULT
+    iocontroller.write_pin(pins["BMS_SENSE"][0], PinValue.LOW)
+    time.sleep(0.1)
+    assert canbus.get_state("air_state") == "FAULT"
+    assert canbus.get_state("air_fault") == "BMS_STATUS"
+
+
+def test_transition_idle_ss_closed(canbus, iocontroller, pins):
+    initialize_to_idle(canbus, iocontroller, pins)
+    reset(iocontroller, pins)
+    time.sleep(0.5)
+    assert canbus.get_state("air_state") == "IDLE"
+
+    iocontroller.write_pin(pins["SS_TSMS"][0], PinValue.LOW)
+
+    time.sleep(0.05)
+    assert "SHUTDOWN_CIRCUIT_CLOSED" == canbus.get_state("air_state")
+
+
+def test_ss_closed_implausibility(canbus, iocontroller, pins):
+    initialize_to_idle(canbus, iocontroller, pins)
+    reset(iocontroller, pins)
+    time.sleep(0.5)
+    assert canbus.get_state("air_state") == "IDLE"
+
+    iocontroller.write_pin(pins["SS_TSMS"][0], PinValue.LOW)
+
+    time.sleep(0.05)
+    assert "SHUTDOWN_CIRCUIT_CLOSED" == canbus.get_state("air_state")
+
+    # Do not close AIR_N_WELD_DETECT
+    time.sleep(0.3)
+
+    assert "FAULT" == canbus.get_state("air_state")
+    assert "SHUTDOWN_IMPLAUSIBILITY" == canbus.get_state("air_fault")
+
+
+def test_transition_ss_closed_precharge(canbus, iocontroller, pins):
+    initialize_to_idle(canbus, iocontroller, pins)
+    reset(iocontroller, pins)
+    time.sleep(0.5)
+    assert canbus.get_state("air_state") == "IDLE"
+
+    iocontroller.write_pin(pins["SS_TSMS"][0], PinValue.LOW)
+    iocontroller.write_pin(pins["AIR_N_WELD_DETECT"][0], PinValue.HIGH)
+
+    time.sleep(0.05)
```

Comment:
I think it would be great to make these delays into constants with a name that explains why these names. Can chat more about this, but I think it keeps the tests way more mainatainable

Reply:
Yeah, I think that _would_ be good. I'm not sure that is the best use of time right now though. A lot of these had to be tweaked a lot. I'm confident in all of the delays in the AIR control firmware, so I think we can punt on this until another time. Thoughts?

Reply:
On one side I do agree that it might feel like not the best use of time, but on the other hand, this file will serve as the basis for all future code, so I think it could actually add high value for the team to replace these in this effort. 
The fact that these needed to be tweaked makes me feel that adding more information in constant names is more critical. We could name the tweaked ones something like EMPIRICAL_<name of delay>